extras/rya.indexing.pcj/src/main/java/org/apache/rya/indexing/pcj/storage/mongo/MongoPcjDocuments.java

```diff
@@ -0,0 +1,451 @@
+            final Set<VisibilityBindingSet> batch = new HashSet<>(1000);
+            while(results.hasNext()) {
+                final VisibilityBindingSet bs = new VisibilityBindingSet(results.next());
+                batch.add( bs );
+                if(batch.size() == 1000) {
+                    addResults(pcjName, batch);
+                    batch.clear();
+                }
+            }
+
+            if(!batch.isEmpty()) {
+                addResults(pcjName, batch);
+            }
+
+        } catch (RepositoryException | MalformedQueryException | QueryEvaluationException e) {
+            throw new PCJStorageException(
+                    "Could not populate a PCJ document with Rya results for the pcj named: " + pcjName, e);
+        }
+    }
+
+    /**
+     * List the document names of the PCJ index tables that are stored in MongoDB
+     * for this instance of Rya.
+     *
+     * @return A list of pcj document names that hold PCJ index data for the current
+     *   instance of Rya.
+     */
+    public List<String> listPcjDocuments() {
+        final List<String> pcjNames = new ArrayList<>();
+
+        //This Bson string reads as:
+        //{} - no search criteria: find all
+        //{ _id: 1 } - only return the _id, which is the PCJ name.
+        final FindIterable<Document> rez = pcjCollection.find((Bson) JSON.parse("{ }, { " + PCJ_ID + ": 1 , _id: 0}"));
+        final Iterator<Document> iter = rez.iterator();
+        while(iter.hasNext()) {
+            pcjNames.add(iter.next().get(PCJ_ID).toString().replace("_METADATA", ""));
+        }
+
+        return pcjNames;
+    }
+
+    /**
+     * Returns all of the results of a PCJ.
+     *
+     * @param pcjName - The PCJ to get the results for. (not null)
+     * @return The authorized PCJ results.
+     */
+    public CloseableIterator<BindingSet> listResults(final String pcjName) {
+        requireNonNull(pcjName);
+
+        // get all results based on pcjName
+        return queryForBindings(new Document(PCJ_NAME, pcjName));
+    }
+
+    /**
+     * Retrieves the stored {@link BindingSet} results for the provided pcjName.
+     *
+     * @param pcjName
+     *            - The pcj to retrieve results for.
+     * @param authorizations
+     *            - The authorizations of the user to restrict results.
+     * @param restrictionBindings
+     *            - The collection of {@link BindingSet}s to restrict results.
+     *            <p>
+     *            Note: the result restrictions from {@link BindingSet}s are an OR
+     *            over ANDS in that: <code>
+     *  [
+     *     bindingset: binding AND binding AND binding,
+     *     OR
+     *     bindingset: binding AND binding AND binding,
+     *     .
+     *     .
+     *     .
+     *     OR
+     *     bindingset: binding
+     *  ]
+     * </code>
+     * @return
+     */
+    public CloseableIterator<BindingSet> getResults(final String pcjName, final Authorizations authorizations,
```

Comment:
The authorizations field isn't being used within this method.

Reply:
because we're not sure how to use it right now.

Reply:
Then why does this method exist at all? It will never behave as it is documented.